pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -307,6 +308,27 @@ Iterable<ArchiveIssue> checkAuthors(String pubspecContent) sync* {
   }
 }
 
+final _preDart3 = VersionConstraint.parse('<3.0.0');
+final _firstDart3Pre = Version.parse('3.0.0').firstPreRelease;
+
+/// Checks if the version range is acceptable by current SDKs.
+Iterable<ArchiveIssue> checkSdkVersionRange(Pubspec pubspec) sync* {
+  final sdk = pubspec.environment?['sdk'];
+  if (sdk == null) return;
+  // We still allow pubspec without SDK constraints.
+  if (sdk.isAny) return;
+
+  // No known version exists.
+  if (sdk.intersect(_preDart3).isEmpty) {
+    yield ArchiveIssue('SDK constraint is outside of current Dart SDKs.');
+  }
+
+  // Dart 3 version accepted with valid upper constraint.
+  if (sdk is VersionRange && sdk.max != null && sdk.allows(_firstDart3Pre)) {
+    yield ArchiveIssue('SDK constraint accepts Dart 3.');
+  }
```

Comment:
Why not simply condense this down to `if (sdk.allows(_firstDart3Pre)) {` ?

Reply:
It will trigger a reject with SDK constraint `>=1.1.0`.

Reply:
With the range requirement this is now just `sdk.allows()`.